Mathlib/Tactic/Simproc/FinsetInterval.lean

```diff
@@ -0,0 +1,196 @@
+    | n + 1 =>
+      let ⟨es, p⟩ ← evalFinsetIccNat m n em q($en - 1)
+      return .done { expr := es, proof? := q(Ico_eq_of_Icc_pred_eq (Nat.succ_ne_zero _) $p) }
+  | _, _, _ => return .continue
+
+/-- Simproc to compute `Finset.Ioc a b` when `a b : ℕ`.
+
+**Warnings**:
+* With the standard depth recursion limit, this simproc can compute intervals of size 250 at most.
+* Make sure to exclude `Finset.insert_eq_of_mem` from your simp call when using this simproc. This
+  avoids a quadratic time performance hit. -/
+simproc_decl Ioc_nat (Ioc _ _) := .ofQ fun u α e ↦ do
+  match u, α, e with
+  | 1, ~q(Finset ℕ), ~q(Ioc (OfNat.ofNat $em) (OfNat.ofNat $en)) =>
+    let some m := em.rawNatLit? | return .continue
+    let some n := en.rawNatLit? | return .continue
+    let ⟨es, p⟩ ← evalFinsetIccNat (m + 1) n q($em + 1) en
+    return .done { expr := es, proof? := q(Ioc_eq_of_Icc_succ_eq $p) }
+  | _, _, _ => return .continue
+
+/-- Simproc to compute `Finset.Ioo a b` when `a b : ℕ`.
+
+**Warnings**:
+* With the standard depth recursion limit, this simproc can compute intervals of size 250 at most.
+* Make sure to exclude `Finset.insert_eq_of_mem` from your simp call when using this simproc. This
+  avoids a quadratic time performance hit. -/
+simproc_decl Ioo_nat (Ioo _ _) := .ofQ fun u α e ↦ do
+  match u, α, e with
+  | 1, ~q(Finset ℕ), ~q(Ioo (OfNat.ofNat $em) (OfNat.ofNat $en)) =>
+    let some m := em.rawNatLit? | return .continue
+    let some n := en.rawNatLit? | return .continue
+    let ⟨es, p⟩ ← evalFinsetIccNat (m + 1) (n - 1) q($em + 1) q($en - 1)
+    return .done { expr := es, proof? := q(Ioo_eq_of_Icc_succ_pred_eq $p) }
+  | _, _, _ => return .continue
+
+/-- Simproc to compute `Finset.Iic b` when `b : ℕ`.
+
+**Warnings**:
+* With the standard depth recursion limit, this simproc can compute intervals of size 250 at most.
+* Make sure to exclude `Finset.insert_eq_of_mem` from your simp call when using this simproc. This
+  avoids a quadratic time performance hit. -/
+simproc_decl Iic_nat (Iic _) := .ofQ fun u α e ↦ do
+  match u, α, e with
+  | 1, ~q(Finset ℕ), ~q(Iic (OfNat.ofNat $en)) =>
+    let some n := en.rawNatLit? | return .continue
+    let ⟨es, p⟩ ← evalFinsetIccNat 0 n q(0) en
+    return .done { expr := es, proof? := q(Iic_eq_of_Icc_zero_eq $p) }
+  | _, _, _ => return .continue
+
+/-- Simproc to compute `Finset.Iio b` when `b : ℕ`.
+
+**Warnings**:
+* With the standard depth recursion limit, this simproc can compute intervals of size 250 at most.
+* Make sure to exclude `Finset.insert_eq_of_mem` from your simp call when using this simproc. This
+  avoids a quadratic time performance hit. -/
+simproc_decl Iio_nat (Iio _) := .ofQ fun u α e ↦ do
+  match u, α, e with
+  | 1, ~q(Finset ℕ), ~q(Iio (OfNat.ofNat $en)) =>
+    let some n := en.rawNatLit? | return .continue
+    match n with
+    | 0 =>
+      return .done { expr := (q(∅) : Q(Finset ℕ)), proof? := q(Iio_zero) }
```

Comment:
```suggestion
      return .done <| .mk _ q(Iio_zero)
```
etc

Reply:
There are two of them which I didn't manage to make Qq-safe. Feel free to push suggestions